OpenRA.Game/Network/Connection.cs

```diff
@@ -78,8 +78,7 @@ public IEnumerable<IPEndPoint> GetConnectEndPoints()
 					{
 						return Enumerable.Empty<IPEndPoint>();
 					}
-				})
-				.ToList();
+				});
```

Comment:
I'm not sure about this. If I read the changes correctly we would iterate over this twice (and get the host addresses - worst case involving a dns query - and create new `IPEndPoint`s twice), once for the `Any` call and the other time in the for loop? (Although `Any` should terminate quickly.)

Reply:
removing the ToList removes a unnecessary casting from a underlying collection to. List only... since no add/remove or other list specific operation is required (only enumeration), there's no point on having it there

Reply:
The idea of `ToList` is/was to "cache" the enumeration. Without it the linq expression will be evaluated anew every time the return value is used.

Reply:
Indeed; this is a very common pattern within OpenRA's code.

Reply:
since the base struct is a array, I refactored back the cache, but as a array and not a list since no add/remove is required here